clientconn.go

```diff
@@ -442,40 +442,28 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		go cc.scWatcher()
 	}
 
-	return cc, nil
-}
-
-// ConnectivityState indicates the state of a client connection.
-type ConnectivityState int
```

Comment:
These changes will break anyone using the connectivity state API (it's new, but, still).

Maybe we can push a PR with only that change in it ASAP while we iterate on this PR?